strutil/core_version.go

```diff
@@ -0,0 +1,48 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+var (
+	coreNameFormat = regexp.MustCompile("^core(?P<version>[0-9]*)$")
+)
+
+func GetCoreVersion(base string) (int, error) {
+	foundCore := coreNameFormat.FindStringSubmatch(base)
+	if foundCore != nil {
+		coreVersionStr := foundCore[coreNameFormat.SubexpIndex("version")]
+		if coreVersionStr == "" {
+			return 16, nil
+		} else {
+			v, err := strconv.Atoi(coreVersionStr)
+			if err != nil {
+				panic(fmt.Sprintf("Unexpected error converting %s to version: %v", base, err))
```

Comment:
(nitpick) I would also avoid the panic() and just return an error, the effect is the same but panic() in library code always feels dangerous (but I know people have different opinions here)

Reply:
Done

Reply:
Also commented it is unreachable code.

Reply:
Thanks, indeed. I am probably a bit too cautious, was thinking that in a future refactor it might not be but then we have tests etc. Oh well, hope I was not overthinking this too much :)